ml-agents/mlagents/trainers/learn.py

```diff
@@ -23,104 +23,218 @@
+        type=int,
+        help="Number of parallel environments to use for training",
+    )
+    parser.add_argument(
+        "--docker-target-name",
+        default=None,
+        dest="docker_target_name",
+        help="Docker volume to store training-specific files",
+    )
+    parser.add_argument(
+        "--no-graphics",
+        default=False,
+        action="store_true",
+        help="Whether to run the environment in no-graphics mode",
+    )
+    parser.add_argument(
+        "--debug",
+        default=False,
+        action="store_true",
+        help="Whether to run ML-Agents in debug mode with detailed logging",
+    )
+    parser.add_argument(
+        "--multi-gpu",
+        default=False,
+        action="store_true",
+        help="Setting this flag enables the use of multiple GPU's (if available) during training",
+    )
+
+    args = parser.parse_args(argv)
+    return CommandLineOptions.from_argparse(args)
+
+
 def run_training(
-    sub_id: int, run_seed: int, run_options: Dict[str, Any], process_queue: Queue
+    sub_id: int, run_seed: int, options: CommandLineOptions, process_queue: Queue
 ) -> None:
     """
     Launches training session.
     :param process_queue: Queue used to send signal back to main.
     :param sub_id: Unique id for training session.
+    :param options: parsed command line arguments
     :param run_seed: Random seed used for training.
     :param run_options: Command line arguments for training.
     """
     # Docker Parameters
-    docker_target_name = (
-        run_options["--docker-target-name"]
-        if run_options["--docker-target-name"] != "None"
-        else None
-    )
-
-    # General parameters
-    env_path = run_options["--env"] if run_options["--env"] != "None" else None
-    run_id = run_options["--run-id"]
-    load_model = run_options["--load"]
-    train_model = run_options["--train"]
-    save_freq = int(run_options["--save-freq"])
-    keep_checkpoints = int(run_options["--keep-checkpoints"])
-    base_port = int(run_options["--base-port"])
-    num_envs = int(run_options["--num-envs"])
-    curriculum_folder = (
-        run_options["--curriculum"] if run_options["--curriculum"] != "None" else None
-    )
-    lesson = int(run_options["--lesson"])
-    fast_simulation = not bool(run_options["--slow"])
-    no_graphics = run_options["--no-graphics"]
-    multi_gpu = run_options["--multi-gpu"]
-    trainer_config_path = run_options["<trainer-config-path>"]
-    sampler_file_path = (
-        run_options["--sampler"] if run_options["--sampler"] != "None" else None
-    )
+    trainer_config_path = options.trainer_config_path
+    curriculum_folder = options.curriculum_folder
 
     # Recognize and use docker volume if one is passed as an argument
-    if not docker_target_name:
-        model_path = "./models/{run_id}-{sub_id}".format(run_id=run_id, sub_id=sub_id)
+    if not options.docker_target_name:
+        model_path = "./models/{run_id}-{sub_id}".format(
```

Comment:
Could consider making these properties of the `CommandLineOptions` class but didn't want to change too much here at once.